### PR TITLE
Add clang instructions for building Win64 julia

### PR DESCRIPTION
**Pull request by Keno**

### CODE REVIEW

File: README.windows.md [hunk 1]
```diff
@@ -117,10 +117,24 @@ or edit `%USERPROFILE%\.gitconfig` and add/edit the lines:
...
+```
+     *N.B.* The path to CMAKE may differ on your system. Adjust accordingly.
+     
+  4. Optionally install clang as your C/C++ compiler
+     1. Download and install the prebuilt clang binaries from http://llvm.org/releases/download.html.
+        The rest of these instructions will assume LLVM was installed to C:\LLVM.
+     2. Set clang as your default compiler:
+     ```
+    cat > Make.user <<EOF
+        override CC_ARG=--sysroot=$(HOME)/julia/usr/x86_64-w64-mingw32/sys-root/mingw -target x86_64-w64-mingw32
```
Comment (tkelman) [inline README.windows.md]: don't assume `$(HOME)/julia`

File: README.windows.md [hunk 2]
```diff
@@ -117,10 +117,24 @@ or edit `%USERPROFILE%\.gitconfig` and add/edit the lines:
...
+        override CXX_ARG=$(CC_ARG)
+        override CC=/C/LLVM/bin/clang.exe $(CC_ARG)
```
Comment (tkelman) [inline README.windows.md]: why does there need to be a separate variable for args?

File: README.windows.md (hunk 1 above)
Comment (tkelman) [inline README.windows.md]: should also separate instructions for 32 vs 64 bit build

**Comment by tkelman** [comment]
I'd rather have this out of line as a separate bullet since it's not widely tested.

File: README.windows.md (hunk 1 above)
Comment (Keno) [inline README.windows.md]: Bullet 1 assumes `$(HOME)/julia`, if people adjust this, that's their own business.

File: README.windows.md (hunk 2 above)
Comment (Keno) [inline README.windows.md]: CMAKE needs CC_BASE and CC_ARG separately. We set CC_BASE automatically based on CC, but the same does not appear to be true for CC_ARG.

**Comment by Keno** [comment]
It's the only configuration that actually works at the moment. Following the instruction as is will get you a gcc that miscompiles LLVM.

**Comment by tkelman** [comment]
I also consider msys2 effectively unsupported, but opensuse has upgraded to gcc 6 which fixed the issue with ipa-ra.

File: README.windows.md (hunk 1 above)
Comment (tkelman) [inline README.windows.md]: only implicitly, if you assume people clone from home. should be more generalizable via pwd

**Comment by tkelman** [comment]
closed because we're not actively supporting msys2 right now. there is a clang-cross package in cygwin these days but libgmp's build system is still broken with it

**Pull request closed without merging**